ddtrace/_trace/span.py

```diff
@@ -486,9 +486,12 @@ def get_metric(self, key: _TagNameType) -> Optional[NumericType]:
     def _add_event(
         self, name: str, attributes: Optional[Dict[str, _JSONType]] = None, timestamp: Optional[int] = None
     ) -> None:
-        """Add an event to the span."""
         self._events.append(SpanEvent(name, attributes, timestamp))
 
+    def _add_on_finish_exception_callback(self, callback: Callable[["Span"], None]):
```

Comment:
🤔  I wonder if we could use the core event hub to notify error tracking. E.g. this is what we listen for in Exception Replay https://github.com/DataDog/dd-trace-py/blob/89b3d7e30267e92fa68a88460342c89de213e7de/ddtrace/_trace/span.py#L551